pkg/distributor/distributorerror/errors.go

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package distributorerror
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+)
+
+const (
+	// 529 is non-standard status code used by some services to signal that "The service is overloaded".
+	StatusServiceOverloaded = 529
+)
+
+// ReplicasNotMatch is an error stating that replicas do not match.
+type ReplicasNotMatch struct {
```

Comment:
Nit, and we can do in another PR, but can we rename this error to `ReplicasDidNotMatch`. I didn't want to comment this in the first place, but since it's the third day reading the same error... :see_no_evil: